src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -7102,73 +7138,100 @@ void StorageReplicatedMergeTree::lockSharedData(const IMergeTreeDataPart & part)
     String id = part.getUniqueId();
     boost::replace_all(id, "/", "_");
 
-    String zookeeper_node = fs::path(zookeeper_path) / zero_copy / "shared" / part.name / id / replica_name;
-
-    LOG_TRACE(log, "Set zookeeper lock {}", zookeeper_node);
-
-    /// In rare case other replica can remove path between createAncestors and createIfNotExists
-    /// So we make up to 5 attempts
-    for (int attempts = 5; attempts > 0; --attempts)
+    Strings zc_zookeeper_paths = getZeroCopyPartPath(*getDefaultSettings(), disk->getType(), getTableUniqID(),
```

Comment:
why we use `getDefaultSettings` here instead of `getSettings`?

Reply:
We need also ability to unfreeze all freezed backup including removed tables, and in this case we will not have any table object and table settings for this kind of parts, only default settings. But now we can use table settings, because unfreeze is possible only for existing tables.
Change for getSettings().